framework/audio/iplayer.h

```diff
@@ -0,0 +1,65 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_IPLAYER_H
+#define MU_AUDIO_IPLAYER_H
+
+#include "async/channel.h"
+
+namespace mu {
+namespace audio {
+class IPlayer
+{
+public:
+    enum Status {
+        Stoped = 0,
+        Running,
+        Error
+    };
+
+    virtual ~IPlayer() = default;
+
+    virtual Status status() const { return m_status; }
+    virtual async::Channel<Status> statusChanged() const { return m_statusChanged; }
+
+    virtual bool isRunning() const { return m_status == Status::Running; }
+
+    virtual void play() = 0;
+    virtual void seek(unsigned long miliseconds) = 0;
+    virtual void stop() = 0;
+
+    virtual unsigned long miliseconds() const = 0;
+    virtual void forwardTime(unsigned long miliseconds) = 0;
+
+protected:
+
+    virtual void setStatus(const Status& status)
+    {
+        if (m_status == status) {
+            return;
+        }
+        m_status = status;
+        m_statusChanged.send(m_status);
+    }
+
+    Status m_status = Status::Stoped;
+    async::Channel<Status> m_statusChanged;
```

Comment:
This class does not look like an interface, so it should not be prefixed with `I`

Reply:
fixed